go/src/koding/klient/machine/dynamic.go

```diff
@@ -192,10 +192,14 @@ func (dc *DynamicClient) cron() {
 
 // tryUpdate uses client builder to ping the machine and updates dynamic client
 // if machine address changes.
-func (dc *DynamicClient) tryUpdate(addr *Addr) {
+func (dc *DynamicClient) tryUpdate(addr *Addr, islog *bool) {
 	stat, a, err := dc.opts.Builder.Ping(dc.opts.AddrFunc)
 	if err != nil {
-		dc.log.Warning("Machine ping error: %s", err)
+		if *islog {
```

Comment:
> there is nothing prevents someone from calling tryUpdate outside of cron() ;) i would accept that argument if it were in cron() as anon/inline func. :raising_hand_man:

Initially it was inlined, I moved it to separate method just to improve readability. And I think that private functions are implementation details to the package so we don't need defensive programming for them. I can find many places in the code base where you did the same [assumptions](https://github.com/koding/koding/blob/f365afd539f663052f5d9c2a60fe2fba749c0436/go/src/koding/workers/removenonexistents/main.go#L229) and, as I said, I don't treat them as errors ([because they are not](https://golang.org/src/net/http/client.go?s=6486:6538#L209)).